mypy/semanal.py

```diff
@@ -3142,12 +3142,14 @@ def process_module_assignment(self, lvals: List[Lvalue], rval: Expression,
             rnode = self.lookup_type_node(rval)
             if rnode and isinstance(rnode.node, MypyFile):
                 for lval in lvals:
-                    if not isinstance(lval, NameExpr):
+                    if not isinstance(lval, RefExpr):
                         continue
                     # respect explicitly annotated type
                     if (isinstance(lval.node, Var) and lval.node.type is not None):
                         continue
                     lnode = self.current_symbol_table().get(lval.name)
+                    if lnode is None and self.type is not None:
```

Comment:
The symbol table selection doesn't work correctly. In the case of the the `self.<var> = x` definitions, the result of `current_symbol_table` returns the `warnings` variable the `import` statement generated when run in the context of the `__init__` function. But we actually want to have `lnode = self.type.names.get(lval.name)`. Is there a better way to get the correct symbol table for the `lval`?

Reply:
Would `lval.node.info.names.get(lval.name)` be a better one?